src/main/java/com/uber/cadence/internal/compatibility/proto/ErrorMapper.java

```diff
@@ -0,0 +1,70 @@
+package com.uber.cadence.internal.compatibility.proto;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+import com.google.rpc.Status;
+import com.uber.cadence.api.v1.*;
+import com.uber.cadence.serviceclient.exceptions.*;
+import io.grpc.StatusRuntimeException;
+import io.grpc.protobuf.StatusProto;
+
+public class ErrorMapper {
+    public static ServiceClientException Error(StatusRuntimeException e) {
+
+        Status status = StatusProto.fromThrowable(e);
+        if (status == null || status.getDetailsCount() == 0) {
+            return new ServiceClientException("empty status or status with empty details", e);
```

Comment:
Nit: When can this happen? What kind of exception does GRPC throw if we have a connection issue rather than an error returned by the server?

Reply:
Server controls whether to pass details. https://github.com/uber/cadence/blob/f4e219ae4248552e53a8ba57036b9af62485221b/common/types/mapper/proto/errors.go#L274
I've simplified the logic to handle such cases.